src/shared/tests.h

```diff
@@ -109,15 +110,27 @@ static inline int run_test_table(void) {
         return r;
 }
 
+static inline int test_nop(void) {
+        return EXIT_SUCCESS;
+}
+
 #define DEFINE_CUSTOM_TEST_MAIN(log_level, intro, outro) \
         int main(int argc, char *argv[]) {               \
-                int _r = EXIT_SUCCESS;                   \
+                int _r, _q;                              \
                 test_setup_logging(log_level);           \
                 save_argc_argv(argc, argv);              \
-                intro;                                   \
-                _r = run_test_table();                   \
-                outro;                                   \
-                return _r;                               \
+                _r = intro();                            \
```

Comment:
Just a comment passing by: if this is turned into something like `_r = intro ? intro() : EXIT_SUCCESS;` the macro could be called with `NULL` as function parameter, which would be more natural than `test_nop` and knowing that not special setup is requested would not require looking up the definition of `test_nop()`.

Reply:
Ooh, I like that.

Reply:
Let's do that later.

Reply:
#22356.